### PR TITLE
RocksJava CI improvements

### PR DESCRIPTION
**Pull request by adamretter**
1. Build the static library for testing on Travis-CI
2. Also build RocksJava on AppVeyor

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1628,6 +1628,12 @@ jtest_run:
 jtest: rocksdbjava
 	cd java;$(MAKE) sample;$(MAKE) test;
 
+jtest_static_compile: rocksdbjavastatic
+	cd java;$(MAKE) java_test
```
Comment (sagar0) [inline Makefile]: Where do you envision this target to be useful?

**Comment by sagar0** [comment]
Appveyor build failed ... could you take a look at that too. Thanks!

File: Makefile (hunk 1 above)
Comment (adamretter) [inline Makefile]: When locally you want to do multiple test runs on the static lib.

**Comment by adamretter** [comment]
@sagar0 Looks like that got broken elsewhere, I just opened a PR to fix that - https://github.com/facebook/rocksdb/pull/2356

**Comment by sagar0** [comment]
Seems like all travis builds failed.

File: .travis.yml [hunk 2]
```diff
@@ -60,7 +60,7 @@ script:
   - if [[ "${TEST_GROUP}" == 'platform_dependent3' ]]; then OPT=-DTRAVIS V=1 ROCKSDBTESTS_START=checkpoint_test ROCKSDBTESTS_END=db_block_cache_test make -j4 check_some; fi
   - if [[ "${TEST_GROUP}" == '1' ]]; then OPT=-DTRAVIS V=1 ROCKSDBTESTS_START=db_block_cache_test ROCKSDBTESTS_END=comparator_db_test make -j4 check_some; fi
   - if [[ "${TEST_GROUP}" == '2' ]]; then OPT=-DTRAVIS V=1 ROCKSDBTESTS_START=comparator_db_test make -j4 check_some; fi
-  - if [[ "${JOB_NAME}" == 'java_test' ]]; then OPT=-DTRAVIS V=1 make clean jclean && make rocksdbjava jtest; fi
+  - if [[ "${JOB_NAME]" == 'java_static_test' ]]; then OPT=-DTRAVIS V=1 make clean jclean && make jtest_static; fi
```
Comment (sagar0) [inline .travis.yml]: s/]/}

Comment (adamretter) [inline .travis.yml]: Hah I also looked at it for sometime and couldn't see it either. Doh!

Thanks :-)

File: Makefile [hunk 3]
```diff
@@ -1471,7 +1473,7 @@ ifeq ($(PLATFORM), OS_SOLARIS)
 	ROCKSDBJNILIB = librocksdbjni-solaris$(ARCH).so
 	ROCKSDB_JAR = rocksdbjni-$(ROCKSDB_MAJOR).$(ROCKSDB_MINOR).$(ROCKSDB_PATCH)-solaris$(ARCH).jar
 	JAVA_INCLUDE = -I$(JAVA_HOME)/include/ -I$(JAVA_HOME)/include/solaris
-	SHA256_CMD = digest -a sha256
+	SHA1_CMD = digest -a sha1
```
Comment (sagar0) [inline Makefile]: I did run into this issue a few weeks back, due to ssl version shipped with OS X being too old. You could continue using sha256 on mac if you switch the command to `openssl sha256 -r` on OSX (or switch to a newer version openssl from homebrew, which we don't need to if we don't want to introduce another dependency). And yes, unfortunately, the cut command also needs to change for OSX :( , as you did. I didn't know the Makefile's code well enough at that time to muster the courage to send out a PR.

Comment (adamretter) [inline Makefile]: @sagar0 Okay cool. I just rebased to remove the merge conflict, I think assuming Travis works, then we should be good to go now.

Comment (sagar0) [inline Makefile]: My question is: Shall we continue using SHA256, as SHA1's pretty old? Thoughts?

Comment (adamretter) [inline Makefile]: @sagar0 Well this is only for the purpose of verifying the checksum of a downloaded file. I don't see any issue with using sha1 for that. Also I don't see sha1 going away anytime soon.

Comment (sagar0) [inline Makefile]: Alright, fine with me.

**Comment by sagar0** [comment]
Seems like appveyor build is still failing for this PR. There are a few more errors even after discounting errors related to SstFileInfo.

**Comment by gfosco** [comment]
Closing this via automation due to lack of activity.  If discussion is still needed here, please re-open or create a new/updated issue.

**Pull request closed without merging**